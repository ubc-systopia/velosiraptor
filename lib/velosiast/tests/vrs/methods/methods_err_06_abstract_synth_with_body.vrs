
// abstract method with body
abstract fn foo() -> bool
{
    true
}

// abstract method with body
synth fn foo() -> bool
{
    true
}
