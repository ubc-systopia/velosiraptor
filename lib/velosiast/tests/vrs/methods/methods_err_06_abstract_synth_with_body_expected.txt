error: method defined as abstract  cannot have a body.
     --> tests/vrs/methods/methods_err_06_abstract_synth_with_body.vrs:3:1
      |
    3 |  abstract fn foo() -> bool
      |  ^^^^^^^^ remove the `abstract ` modifier
error: could not compile due to 1 previous errors; 0 warnings emitted


@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@

error: method defined as synth  cannot have a body.
     --> tests/vrs/methods/methods_err_06_abstract_synth_with_body.vrs:9:1
      |
    9 |  synth fn foo() -> bool
      |  ^^^^^ remove the `synth ` modifier
error: could not compile due to 1 previous errors; 0 warnings emitted


@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@

